internal/scorecard/alpha/scorecard.go

```diff
@@ -100,12 +116,22 @@ func selectTests(selector labels.Selector, tests []Test) []Test {
 }
 
 // runTest executes a single test
-func (o Scorecard) runTest(test Test) (result *v1.Pod, err error) {
+func (o Scorecard) runTest(_ context.Context, test Test) (result v1alpha2.ScorecardTestResult, err error) {
 
 	// Create a Pod to run the test
 	podDef := getPodDefinition(test, o)
-	result, err = o.Client.CoreV1().Pods(o.Namespace).Create(context.TODO(), podDef, metav1.CreateOptions{})
-	return result, err
+	pod, err := o.Client.CoreV1().Pods(o.Namespace).Create(context.TODO(), podDef, metav1.CreateOptions{})
+	if err != nil {
+		return result, err
+	}
+
+	err = o.waitForTestToComplete(pod)
+	if err != nil {
+		return result, err
+	}
+
+	result = getTestResult(o.Client, pod, test)
```

Comment:
Pass the context down to this function to be used.

Also, can this be `o.getTestResult(ctx, pod, test)`?

Reply:
done and updated the function sig.